libsupport/include/katana/EntityTypeManager.h

```diff
@@ -78,6 +83,181 @@ class EntityTypeManager {
         atomic_entity_type_id_to_entity_type_ids_(
             std::move(atomic_entity_type_id_to_entity_type_ids)) {}
 
+  template <typename ArrowType>
+  struct PropertyColumn {
+    int field_index;
+    std::shared_ptr<ArrowType> array;
+
+    PropertyColumn(int i, std::shared_ptr<ArrowType>& a)
+        : field_index(i), array(a) {}
+  };
+
+  /// This function can be used to convert "old style" graphs (storage format 1,
+  /// where types are represented by bool or uint8 properties) and "new style"
+  /// graphs (version > 2, where types are represented in our native type
+  /// represenation). This function is serial but it likely iterates over
+  /// O(nodes) and O(edges) vectors, so it is very slow. It should only be used
+  /// for updating old graphs.
+  ///
+  /// The number of entries in entity_type_ids should be equal to topo_size and
+  /// properties->num_rows().
+  template <template <typename> class Vector>
```

Comment:
Is a template parameter required? If so, consider instantiating the specific ones required. This function is big enough that it's better defined in a cpp file. You can do this in a subsequent PR.

Reply:
The template was a workaround for handling `NUMAArray`. I don't think we need to support arrays other than `NUMAArray`, but as long as we define this in `libsupport` I don't think we can instantiate that instance.

Reply:
I understand the point. I'm a bit concerned about increased compile time because this header will be included in many places. Can we instead take a C-style pointer to represented an array? I'm not sure if that is better for code quality but we don't expect this function to be used a lot anyway.

Reply:
You know, we could solve this issue by introducing a `Buffer` type that takes a pointer and acts like a vector/array and has a size. Pretty much an Arrow Buffer (but without the unnecessary shared_ptr).

Reply:
This function is currently only called from PropertyGraph.cpp. I honestly think it should probably be there. It's pretty coupled to PropertyGraph loading.

Reply:
Amber, that's a good idea.

Arthur, no, it needs to be called when the RDG is loaded. That PR will come next.

Reply:
I will add a `Buffer` class to libsupport in my follow-up PR.

Reply:
I would like to see benchmarks that show the downside of using arrow::Array before we create our own copy. I suspect the overhead of `shared_ptr` is lower than people think in any real context. I may well be wrong. But adding our own copies of things before benchmarking seems silly and a good way to force ourselves to reimplement more and more of arrow.

Reply:
@arthurp Are you suggesting that we take an `arrow::Array` as the parameter here?

Reply:
I was thinking `arrow::Buffer` but `arrow::Array` would be fine too in terms of what I'm thinking about.

Reply:
Is `arrow::Buffer` designed to be mutable?

Reply:
Yes.

Reply:
Does `arrow::Buffer` have the coveted `operator []`? Secondly, if I don't need a `shared_ptr`, then I shouldn't use it. I just need to wrap a pointer into a `Buffer`. `Buffer` never owns the memory nor frees it. So I'm in favor of adding our own simple `Buffer` types and I'll even volunteer to add one. I've had this itch for a while.

Reply:
I looked it up and `arrow::Buffers` are untyped, so totally unfit for our use case here. I want something typed that behaves like an array or vector.